README.md

```diff
@@ -1,3 +1,251 @@
+```
+
+#### Running
+
+```
+rspec spec
+```
+
+---
+
+### Logging
+
+Standard Rails logging still applies, but whenever an API returns an error message, a log is created in the `/log/api.errors.<ENVIRONMENT>.log` file.
+
+Each log will have the following JSON message:
+
+- `when`: datetime, The date and time the error occurred
+- `what`: string, The message of the error
+- `where`: string, The API path being called when the error occurred
+- `who`: Object that contains more information of the error
+  - `url`: string, The URL of the call when the error occurred
+  - `verb`: string, The request method (GET, POST, DELETE, UPDATE)
+  - `parameters`: object, The parameters passed to the call
+- `how`: string, The backtrace of the error. This will only log for the `development` environment.
+
+---
+
+## Development Notes
+
+I made to sure to implement a basic versioning system from the start (`/v1`) to ensure backwards compatability and that legacy apps continue to work in case new versions change the response or logical biz rules.
+
+I chose to have the `api_key` read from the standard parameters rather than the header pretty much for ease of implementation. Ideally the clients using the API should not expose their keys to the public, meaning that they should not make calls directly from their browser apps but instead from their own secured servers. If the API is deemed to require much more security, we can make changes to require that the `api_key` be passed only via the header.
+
+Additional Biz Rules:
+
+- When restrictions are newly added (ie restaurant not allowed to deliver to a specific zone), future
+
+Assumptions:
+
+- Assignments are created on an on demand basis. Meaning, there are no automatic creations of assignments. Assignments can be created in the admin for a day at a time.
+- When Restaurants are deleted, all dependent associations are also removed. This includes Meals, Restriction Rules, and Assignments either in the past, present, or future.
+- The `api_key` for the API is bound to a Restaurant object. Thus I have Assignments being returned only for that Restaurant. I'm assuming that we'd want to enforce some kind of privacy to not allow Restaurants view the data of other Restaurants.
+
+If I had more time or more freedom to increase the scope of this project:
+
+- Currently, deleting resources actually deletes records from the database. Ideally, I'd preserve data as much as possible. So if a Restaurant should be deleted, I'd make it "inactive" with a new column flag like `is_active` and set it to `false`. Then this resource shouldn't be used in the future or viewed in the admin. The most relevant business use case for this is to view history of deliveries in the past for historical purposes.
+- Much more comprehensive rspec tests. Right now, I focused a majority of time when creating tests around the biz rules for creating assignments as well as the API. There are no tests for the main admin at this time.
```

Comment:
It would be nice if you had some specs to cover these views. Perhaps you are better than me at manual testing, but I never felt confident with anything I wrote until I have complete spec coverage :sigh..

Reply:
No, you are right. It's much better to have spec coverage than doing manual testing. Especially when the project grows larger and larger, with more complex biz rules being added all the time.